src/test/java/com/jfrog/ide/common/yarn/YarnTreeBuilderTest.java

```diff
@@ -119,4 +121,71 @@ private void fourChildrenScenario() {
         }
         assertEquals(count, 4);
     }
+
+    @Test
+    public void extractMultiplePathsTest() {
+        String projectRootId = "root";
+        String packageFullName = "pkg:1.0.0";
+        List<String> rawDependencyPaths = List.of(
+                "Specified in \"dependencies\"",
+                "Hoisted from \"jest-cli#node-notifier#minimist\"",
+                "Hoisted from \"jest-cli#sane#minimist\"",
+                "Hoisted from \"@test0#\\test1#test2#pkg\"",
+                "This module exists because \"jest-cli#istanbul-api#mkdirp\" depends on it.",
+                "This module exists because it's specified in \"devDependencies\"."
+        );
+
+        List<List<String>> expectedPaths = List.of(
+                List.of(projectRootId, packageFullName),
+                List.of(projectRootId, "jest-cli", "node-notifier", "minimist", packageFullName),
+                List.of(projectRootId, "jest-cli", "sane", "minimist", packageFullName),
+                List.of(projectRootId, "@test0", "\\test1", "test2", packageFullName),
+                List.of(projectRootId, "jest-cli", "istanbul-api", "mkdirp", packageFullName),
+                List.of(projectRootId, packageFullName)
+        );
+        YarnTreeBuilder yarnTreeBuilder = new YarnTreeBuilder(Paths.get(""), "", null, new NullLog());
+        List<List<String>> paths = yarnTreeBuilder.extractMultiplePaths(projectRootId, packageFullName, rawDependencyPaths);
```

Comment:
To emphasize the different between the "expected"
```suggestion
        List<List<String>> actualPaths = yarnTreeBuilder.extractMultiplePaths(projectRootId, packageFullName, rawDependencyPaths);
```